app/code/Magento/InventoryInStorePickup/Model/Carrier/InStorePickup.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\Carrier;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\InventoryInStorePickup\Model\Carrier\Command\GetFreeBoxes;
+use Magento\InventoryInStorePickup\Model\Carrier\Command\GetIsAnyPickupLocationAvailable;
+use Magento\InventoryInStorePickup\Model\Carrier\Command\GetShippingPriceInterface;
+use Magento\Quote\Model\Quote\Address\RateRequest;
+use Magento\Quote\Model\Quote\Address\RateResult\ErrorFactory;
+use Magento\Quote\Model\Quote\Address\RateResult\Method;
+use Magento\Quote\Model\Quote\Address\RateResult\MethodFactory;
+use Magento\Shipping\Model\Carrier\AbstractCarrier;
+use Magento\Shipping\Model\Carrier\CarrierInterface;
+use Magento\Shipping\Model\Rate\ResultFactory;
+use Psr\Log\LoggerInterface;
+
+/**
+ * In-Store Pickup Delivery Method Model.
+ */
+class InStorePickup extends AbstractCarrier implements CarrierInterface
```

Comment:
Probably, need to be moved to Magento_InventoryInStorePickupApi

Reply:
Let's find a way to avoid all of these dependencies in API